ci/deps/actions-310.yaml

```diff
@@ -31,8 +31,7 @@ dependencies:
   - jinja2
   - lxml
   - matplotlib
-  # TODO: uncomment after numba supports py310
-  #- numba
+  - numba
```

Comment:
This could cause installing an older numpy version which could(?) explain most of the errors (but not the pyqt stuff).

Reply:
This file should only affect our specific PY 3.10 build which just runs the unit tests though. The typing checks should have an environment that is set up by `environment.yml`

Reply:
Sorry, I meant that since we anyways run the unit tests with NumPy-dev in a separate workflow, prioritizing the latest numba version over the latest (released) NumPy version (in `environment.yml`) could be fine. Either way, it would be good to limit the numba version or the numpy version in `environment.yml`.

Reply:
> As there is one dedicated CI run for numpy-dev, it would make sense to use the latest numpy compatible with numba (even for typing). Reverting #45244 would probably fix most of the numpy errors.

The problem I have with this is that new contribritors when setting up an environment will get the latest numpy and have mypy errors by default.

We should make the contributor experience pain free so (imo) we should use environment.yaml for the typing validation to match the local dev env .

Otherwise, this just makes it difficult for people to contribute to the typing issues.

Now, numba is included in environment.yaml so I'm not sure why when I set up a clean dev locally I get numpy 1.23.1 and on ci we get 1.22.4 (maybe there is some caching on ci?)

My comments here are from looking into this a couple of weeks ago. So this comment here now maybe out of date. Will look again soon.

Reply:
> Now, numba is included in environment.yaml so I'm not sure why when I set up a clean dev locally I get numpy 1.23.1 and on ci we get 1.22.4 (maybe there is some caching on ci?)

I must admit that I don't use the official way to setup a pandas-dev env, but it would be great to ensure that the officially documented pandas-env does not cause mypy errors.

Maybe numba has different numpy-constraints on conda-forge (or conda installs incompatible versions)? When I ask poetry to install `numba = ">=0.53.1"` (as in environment.yml) and `numpy = ">=1.23.0"`, it is unable to find a solution (at least not on Linux with python 3.10).

Reply:
> I must admit that I don't use the official way to setup a pandas-dev env, but it would be great to ensure that the officially documented pandas-env does not cause mypy errors.

yes I need to double check that's still true.